application/controllers/SurveyAdministrationController.php

```diff
@@ -2632,7 +2615,7 @@ public function changeTemplate($iSurveyID, $template, $aResults = null, $bReturn
 
         $survey = Survey::model()->findByPk($iSurveyID);
 
-        if (!Permission::model()->hasSurveyPermission($iSurveyID, 'surveyactivation', 'update')) {
+        if (!Permission::model()->hasSurveyPermission($iSurveyID, 'surveysettings', 'update')) {
             if (!empty($bReturn)) {
```

Comment:
See https://github.com/LimeSurvey/LimeSurvey/blob/0b3a1c0bcab8c5359595b027eafa2e0238318126/application/controllers/SurveyAdministrationController.php#L789 it's an empty array, then break if 1st survey don't have valid permission.

`!is_array` maybe ? Or isnull ?

Reply:
why? not following. Please notice this was already working... we only changed the permission checking...

Reply:
Yes but before user have all rights (it was superadmin) then permission return always true this part was never used .

`$aResults` start by an empty array, then if user don't have permission on the 1st surey : it die … https://github.com/LimeSurvey/LimeSurvey/blob/0b3a1c0bcab8c5359595b027eafa2e0238318126/application/controllers/SurveyAdministrationController.php#L2625

Reply:
No, it is not die-ing :)
You can see `$bReturn` is sent as true sometimes, which returns the errors on an array.